src/main/java/org/reaktivity/nukleus/socks/internal/stream/Correlation.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.socks.internal.stream;
+
+import java.util.function.Consumer;
+
+import org.reaktivity.nukleus.function.MessageConsumer;
+import org.reaktivity.nukleus.socks.internal.types.control.RouteFW;
+
+public class Correlation
+{
+
+    // Can be used to send RESET and WINDOW back to the SOURCE on the ACCEPT stream
+    private MessageConsumer acceptThrottle;
+
+    // ACCEPT stream identifier
+    private long acceptStreamId;
+
+    // ACCEPT SOURCE reference (similar to port)
+    private long acceptSourceRef;
+
+    // ACCEPT SOURCE name (similar to address)
+    private String acceptSourceName;
+
+    // Can be used to send BEGIN, DATA, END, ABORT frames to the SOURCE on the ACCEPT-REPLY stream
```

Comment:
This data will probably fall out of date quickly as more frame types are implemented...